### PR TITLE
Adding and improving function documentation for `ADODB_mysqli`

### PR DESCRIPTION
**Pull request by wonderboyjon**
Related to #488

### CODE REVIEW

File: drivers/adodb-mysqli.inc.php [hunk 1]
```diff
@@ -83,14 +90,25 @@ function SetTransactionMode( $transaction_mode )
...
-	// To add: parameter int $port,
-	//         parameter string $socket
-	function _connect($argHostname = NULL,
-				$argUsername = NULL,
-				$argPassword = NULL,
-				$argDatabasename = NULL, $persist=false)
-	{
+    /**
+     * Connect to a database.
+     * TODO add: parameter int $port, parameter string $socket
```
Comment (dregad) [inline drivers/adodb-mysqli.inc.php]: Use `@todo` tag

File: drivers/adodb-mysqli.inc.php [hunk 2]
```diff
@@ -153,28 +171,64 @@ function _connect($argHostname = NULL,
...
 
-	function IfNull( $field, $ifNull )
-	{
-		return " IFNULL($field, $ifNull) "; // if MySQL
-	}
-
-	// do not use $ADODB_COUNTRECS
-	function GetOne($sql,$inputarr=false)
+    /**
+     * Returns a stringing containing a MySQL IFNULL() function call for the provided values.
```
Comment (dregad) [inline drivers/adodb-mysqli.inc.php]: stringing -> string ?

File: drivers/adodb-mysqli.inc.php [hunk 3]
```diff
@@ -153,28 +171,64 @@ function _connect($argHostname = NULL,
...
+     *
+     * @return string
+     */
+    function IfNull($field, $ifNull)
+    {
+        return " IFNULL($field, $ifNull) ";
+    }
+
+    /**
+     * Do not use $ADODB_COUNTRECS.
```
Comment (dregad) [inline drivers/adodb-mysqli.inc.php]: I know you just used whatever was there, but this comment is very confusing. Might as well remove it.

File: drivers/adodb-mysqli.inc.php [hunk 4]
```diff
@@ -323,7 +424,13 @@ function CreateSequence($seqname='adodbseq',$startID=1)
 		return $this->Execute(sprintf($this->_genSeq2SQL,$seqname,$startID-1));
 	}
 
-	function GenID($seqname='adodbseq',$startID=1)
+    /**
+     * @param string $seqname
```
Comment (dregad) [inline drivers/adodb-mysqli.inc.php]: Missing description

File: drivers/adodb-mysqli.inc.php [hunk 5]
```diff
@@ -946,7 +1158,7 @@ function SetCharSet($charset_name)
 }
 
 /*--------------------------------------------------------------------------------------
-	 Class Name: Recordset
+	 Class Name: ADORecordSet_mysqli
 --------------------------------------------------------------------------------------*/
```
Comment (dregad) [inline drivers/adodb-mysqli.inc.php]: This should be a PHPDoc block too; and consider adding one for class _ADODB_mysqli_ too.
```
/**
 * Class ADORecordSet_mysqli
 */
```

**Comment by dregad** [comment]
As a side note, when submitting pull requests I strongly advise not to use your _master_ branch, but create a dedicated feature branch instead.

**Comment by wonderboyjon** [comment]
Yeh I realised shortly afterwards that not creating my own branches was a bad idea. Will rectify that after this request.

I've gone through and made all the changes suggested. I tried to marry up the function calls to the website documentation where possible.

[Fields()](https://adodb.org/dokuwiki/doku.php?id=v5:reference:recordset:fields) was an odd one though, as the website documentation makes it clear that the function parameter is expected to be an `int`. However the code for that function in `ADORecordSet_mysqli` very clearly expects a `string`, so I have chosen to respect what the code does and documented it as a string rather than following the website documentation. If this is incorrect let me know.

**Pull request closed without merging**